src/time.cr

```diff
@@ -605,10 +605,18 @@ struct Time
     {% end %}
 
     unless offset
-      # current TZ may have DST, either in past, present or future
-      ret = LibC.localtime_r(pointerof(second), out tm)
-      raise Errno.new("localtime_r") if ret.null?
-      offset = tm.tm_gmtoff.to_i64
+      {% if !flag?(:windows) %}
+        # current TZ may have DST, either in past, present or future
+        ret = LibC.localtime_r(pointerof(second), out tm)
+        raise Errno.new("localtime_r") if ret.null?
+        offset = tm.tm_gmtoff.to_i64
+      {% else %}
+        # TODO handle daylight?
+        # https://msdn.microsoft.com/en-us/library/windows/desktop/ms724421(v=vs.85).aspx
+        LibWindows.get_time_zone_information(out tz)
```

Comment:
Unknown should not raise.

> TIME_ZONE_ID_UNKNOWN: Daylight saving time is not used in the current time zone, because there are no transition dates or automatic adjustment for daylight saving time is disabled.

Invalid yes.

> If the function fails for other reasons, such as an out of memory error, it returns TIME_ZONE_ID_INVALID. To get extended error information, call GetLastError.